dataset/dataset-import.sh

```diff
@@ -1,92 +1,130 @@
 #!/bin/bash
 
 set -e
+set -o pipefail
 
 ### DATASET PROVIDER - KEYCLOAK REST API SERVICE ###
 
-create_realms () {
-  if [ $# -eq 0 ]
-  then
-    REALM_COUNT=1
-  else
-    REALM_COUNT=$1
-  fi
-  CLIENT_COUNT=${2:-100}
-  USER_COUNT=${3:-100}
-  PASSWORD_HASHING=${4:-20000}
-  KEYCLOAK_URI="${5:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
-
-  echo "Creating $REALM_COUNT realm/s with $CLIENT_COUNT client/s and $USER_COUNT user/s with $PASSWORD_HASHING password hash iterations."
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-realms?count=$REALM_COUNT&clients-per-realm=$CLIENT_COUNT&users-per-realm=$USER_COUNT&password-hash-iterations=$PASSWORD_HASHING"
-    echo ""
+set_environment_variables () {
+  ACTION="help"
+  REALM_COUNT="1"
+  REALM_NAME="realm-0"
+  CLIENTS_COUNT="100"
+  USERS_COUNT="100"
+  EVENTS_COUNT="100"
+  SESSIONS_COUNT="100"
+  HASH_ITERATIONS="20000"
+  KEYCLOAK_URI="https://keycloak.$(minikube ip).nip.io/realms/master/dataset"
+  REALM_PREFIX="realm"
+  STATUS_TIMEOUT="120"
+
+  while getopts ":a:r:n:c:u:e:o:i:p:l:t:" OPT
+  do
+    case $OPT in
+      a)
+        ACTION=$OPTARG
+        ;;
+      r)
+        REALM_COUNT=$OPTARG
+        ;;
+      n)
+        REALM_NAME=$OPTARG
+        ;;
+      c)
+        CLIENTS_COUNT=$OPTARG
+        ;;
+      u)
+        USERS_COUNT=$OPTARG
+        ;;
+      e)
+        EVENTS_COUNT=$OPTARG
+        ;;
+      o)
+        SESSIONS_COUNT=$OPTARG
+        ;;
+      i)
+        HASH_ITERATIONS=$OPTARG
+        ;;
+      p)
+        REALM_PREFIX=$OPTARG
+        ;;
+      l)
+        KEYCLOAK_URI=$OPTARG
+        ;;
+      t)
+        STATUS_TIMEOUT=$OPTARG
+        ;;
+      ?)
+        echo "Invalid option: $OPT, read the usage carefully -> "
+        help
+        exit 1
+    esac
+  done
+}
 
+create_realms () {
+  echo "Creating $1 realm/s with $2 client/s and $3 user/s with $4 password hash iterations."
+  execute_command "create-realms?count=$1&clients-per-realm=$2&users-per-realm=$3&password-hash-iterations=$4"
 }
 
 create_clients () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 client/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-clients?count=$1&realm-name=$2"
+  execute_command "create-clients?count=$1&realm-name=$2"
 }
 
 create_users () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 user/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-users?count=$1&realm-name=$2"
+  execute_command "create-users?count=$1&realm-name=$2"
```

Comment:
create-users seems to have regression originating from the DatasetResourceProvider.java, but I dont think its related to any changes here., this is merely an FYI. I will raise a different GH ticket for the issue I observed.

```
	
{"timestamp":"2023-01-11T21:29:42.844Z","sequence":5548,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.benchmark.dataset.DatasetResourceProvider","level":"ERROR","message":"unable to complete task","threadName":"Thread-9","threadId":86,"mdc":{},"ndc":"","hostName":"keycloak-0","processName":"QuarkusEntryPoint","processId":1,"exception":{"refId":1,"exceptionType":"java.util.concurrent.ExecutionException","message":"java.lang.ArithmeticException: / by zero","frames":[{"class":"java.util.concurrent.FutureTask","method":"report","line":122},{"class":"java.util.concurrent.FutureTask","method":"get","line":191},{"class":"org.keycloak.benchmark.dataset.ExecutorHelper","method":"waitForAllToFinish","line":67},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"createUsersImpl","line":393},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"lambda$createUsers$10","line":362},{"class":"java.lang.Thread","method":"run","line":833}],"causedBy":{"exception":{"refId":2,"exceptionType":"java.lang.ArithmeticException","message":"/ by zero","frames":[{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"createUsers","line":999},{"class":"org.keycloak.benchmark.dataset.DatasetResourceProvider","method":"lambda$addUserCreationTasks$11","line":415},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"lambda$runJobInTransaction$1","line":256},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransactionWithResult","line":269},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransaction","line":255},{"class":"org.keycloak.models.utils.KeycloakModelUtils","method":"runJobInTransactionWithTimeout","line":359},{"class":"org.keycloak.benchmark.dataset.ExecutorHelper","method":"lambda$addTaskRunningInTransaction$0","line":58},{"class":"java.util.concurrent.Executors$RunnableAdapter","method":"call","line":539},{"class":"java.util.concurrent.FutureTask","method":"run","line":264},{"class":"java.util.concurrent.ThreadPoolExecutor","method":"runWorker","line":1136},{"class":"java.util.concurrent.ThreadPoolExecutor$Worker","method":"run","line":635},{"class":"java.lang.Thread","method":"run","line":833}]}}}}
```

Reply:
yes most probably caused by this commit: https://github.com/keycloak/keycloak-benchmark/commit/d6d55d4b017598e247597ec5d32cdb19351890ed